pkg/controllers/controller.go

```diff
@@ -70,17 +73,22 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		}
 		return reconcile.Result{}, err
 	}
-	// 2. Copy object for merge patch base
+	// 2. Sanity check, validate the object
+	var a webhook.Validator = c.For()
+	if err := a.ValidateCreate(); err != nil {
+		return reconcile.Result{}, err
```

Comment:
Does this return a sane error message without annotating? I don't think we want to return error, or it will loop. Maybe just log and return result,nil